### PR TITLE
 adding Azure Redis Cache management commands for XPlat CLI 

### PR DESCRIPTION
**Pull request by alfantp**

### CODE REVIEW

**Comment by azurecla** [comment]
Hi __@alfantp__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <p>
        It looks like you're working at Microsoft (alfantp). If you're full-time, we DON'T require a contribution license agreement.
    </p>
    <p>
        If you are a vendor, or work for Microsoft Open Technologies, DO please sign the electronic contribution license agreement. It will take 2 minutes and there's no faxing! https://cla.azure.com.
    </p>

TTYL, AZPRBOT;

File: lib/commands/arm/rediscache/rediscache._js [hunk 1]
```diff
@@ -0,0 +1,742 @@
...
+    .option('-e, --enable-non-ssl-port', $('EnableNonSslPort property of the Redis Cache. Add this flag if you want to enable the Non SSL Port for your cache'))
+    .option('-s, --subscription <id>', $('the subscription identifier'))
+    .execute(function (name, resourceGroup,location, options, _) {
+      
+    ///////////////////////
+    // Parse arguments.  //
+    ///////////////////////
+          
+    log.verbose('arguments: ' + JSON.stringify({
+    name: name,
```
Comment (amarzavery) [inline lib/commands/arm/rediscache/rediscache._js]: indent 2 spaces

File: lib/commands/arm/rediscache/rediscache._js [hunk 2]
```diff
@@ -0,0 +1,742 @@
...
+    options: options
+    }));
+    
+    options.name = options.name || name
+    options.resourceGroup = options.resourceGroup || resourceGroup
+    options.location = options.location || location
+    
+        
+    if (!options.name) {
+    return cli.missingArgument('name');
```
Comment (amarzavery) [inline lib/commands/arm/rediscache/rediscache._js]: indent 2 spaces

File: lib/commands/arm/rediscache/rediscache._js [hunk 3]
```diff
@@ -0,0 +1,742 @@
...
+    } catch(e) {
+        if(!e.code === 'ResourceNotFound' || !e.code === 'ResourceGroupNotFound') {
+        throw e;
+      }
+    }
+    finally {
+    checkRedisCacheProgress.end();
+    }
+        
+    if(redisCacheNameExists){
```
Comment (amarzavery) [inline lib/commands/arm/rediscache/rediscache._js]: if(redisCacheNameExists)space{

File: lib/commands/arm/rediscache/rediscache._js [hunk 4]
```diff
@@ -0,0 +1,742 @@
...
+      throw new Error(util.format($('The requested cache name is unavailable: %s'), options.name));
+      }
```
Comment (amarzavery) [inline lib/commands/arm/rediscache/rediscache._js]: should be out by 2 spaces

File: lib/commands/arm/rediscache/rediscache._js [hunk 5]
```diff
@@ -0,0 +1,742 @@
...
+    ///////////////////////
+    // Parse arguments.  //
+    ///////////////////////
+          
+    log.verbose('arguments: ' + JSON.stringify({
+    name: name,
+    options: options
+    }));
+    
+    options.name = options.name || name
```
Comment (amarzavery) [inline lib/commands/arm/rediscache/rediscache._js]: missing semicolon

File: lib/commands/arm/rediscache/rediscache._js [hunk 6]
```diff
@@ -0,0 +1,742 @@
...
+    var client = utils.createRedisCacheManagementClient(subscription);
+    
+    /////////////////////////////
+    // Check if tenant exists. //
+    /////////////////////////////
+    
+    var checkRedisCacheProgress = cli.interaction.progress(util.format($('')));
+    var redisCacheNameExists = null;
+    try {
+      var redisCacheNameExists = client.redis.get(options.resourceGroup, options.name, _).resource;
```
Comment (amarzavery) [inline lib/commands/arm/rediscache/rediscache._js]: you already declared redisCacheNameExists on line 118

File: lib/commands/arm/rediscache/rediscache._js [hunk 7]
```diff
@@ -0,0 +1,742 @@
...
+    } catch(e) {
+        if(!e.code === 'ResourceNotFound' || !e.code === 'ResourceGroupNotFound') {
```
Comment (amarzavery) [inline lib/commands/arm/rediscache/rediscache._js]: if(!(e.code === 'ResourceNotFound' || e.code === 'ResourceGroupNotFound')) is much easier tounderstand

**Comment by amarzavery** [comment]
please resolve the jshint issues..
https://travis-ci.org/Azure/azure-xplat-cli/jobs/77076380

**Pull request merged**